### PR TITLE
feat(node): make nats optional

### PR DESCRIPTION
**Pull request by namn-grg**
Make NATS optional and not a hard requirement. If NATS URL is provided the events are printed via stdout and also written to log file

### CODE REVIEW

File: pkg/ethereum/nats.go [hunk 1]
```diff
@@ -77,6 +128,21 @@ func (d *DiscoveryV5) sendPeerEvent(ctx context.Context, node *enode.Node, hInfo
...
+		Str("enr", node.String()).
+		Str("id", hInfo.ID.String()).
+		Str("ip", hInfo.IP).
+		Int("port", hInfo.Port).
+		Str("crawler_id", peerEvent.CrawlerID).
+		Str("crawler_location", peerEvent.CrawlerLoc).
+		Msg("peer_discovered event")
+
+	if d.js == nil {
+		fmt.Fprintf(d.fileLogger, "%s ENR: %s, ID: %s, IP: %s, Port: %d, CrawlerID: %s, CrawlerLoc: %s\n",
```
Comment (mempirate) [inline pkg/ethereum/nats.go]: Let's print the stringified json of the event for consistency: `fmt.Fprintln(n.fileLogger, string(json))`

File: pkg/ethereum/nats.go [hunk 2]
```diff
@@ -30,10 +34,57 @@ type MetadataReceivedEvent struct {
...
+		Str("multiaddr", event.Multiaddr).
+		Uint("epoch", event.Epoch).
+		Any("metadata", event.MetaData).
+		Str("client_version", event.ClientVersion).
+		Str("crawler_id", event.CrawlerID).
+		Str("crawler_location", event.CrawlerLoc).
+		Msg("metadata_received event")
+
+	if n.js == nil {
+		fmt.Fprintf(n.fileLogger, "%s ID: %s, Multiaddr: %s, Epoch: %d, Metadata: %v, ClientVersion: %s, CrawlerID: %s, CrawlerLoc: %s\n",
```
Comment (mempirate) [inline pkg/ethereum/nats.go]: Same as above

**Review by mempirate — CHANGES_REQUESTED** [review]
LGTM, some small nits on output format

File: config/config.go [hunk 3]
```diff
@@ -64,7 +65,7 @@ var DefaultDiscConfig DiscConfig = DiscConfig{
 	TCP:        9000,
 	DBPath:     "",
 	ForkDigest: [4]byte{0x6a, 0x95, 0xa1, 0xa9},
-	LogPath:    "nodes.log",
+	LogPath:    "peer_discovered.log",
```
Comment (namn-grg) [inline config/config.go]: Changed path to be consistent with event name

File: config/config.go [hunk 4]
```diff
@@ -113,4 +116,5 @@ var DefaultNodeConfig NodeConfig = NodeConfig{
 	ConcurrentDialers: 64,
 	IP:                "0.0.0.0",
 	Port:              9000,
+	LogPath:           "metadata_received.log",
```
Comment (namn-grg) [inline config/config.go]: Same here

File: main.go [hunk 5]
```diff
@@ -61,7 +61,7 @@ func main() {
 				Name:        "nats-url",
 				Usage:       "NATS server URL (needs JetStream)",
 				Aliases:     []string{"n"},
-				Value:       "nats://localhost:4222",
+				Value:       "", // If empty URL, run without NATS
```
Comment (namn-grg) [inline main.go]: We want Valtrack to be functional without NATS also. Assuming here if `nats-url` flag isn't set the user want to run without it.

File: pkg/ethereum/nats.go [hunk 6]
```diff
@@ -32,10 +36,49 @@ type MetadataReceivedEvent struct {
 	Timestamp         int64           `json:"timestamp"` // Timestamp in UNIX milliseconds
 }
 
+func createNatsStream(url string) (js jetstream.JetStream, err error) {
+	// If empty URL, return nil and run without NATS
+	if url == "" {
+		return nil, nil
+	}
```
Comment (namn-grg) [inline pkg/ethereum/nats.go]: If no NatsURL then return `nil` jetstream

File: pkg/ethereum/node_notifiee.go [hunk 7]
```diff
@@ -179,12 +170,6 @@ func (n *Node) handleInboundConnection(pid peer.ID) {
 	event := info.IntoMetadataEvent()
 
 	n.sendMetadataEvent(ctx, event)
-
-	json, _ := json.Marshal(event)
-
-	n.log.Info().Msgf("Succesful handshake: %s", string(json))
-
-	fmt.Fprintln(n.fileLogger, string(json))
```
Comment (namn-grg) [inline pkg/ethereum/node_notifiee.go:187]: Doing this inside `sendMetadataEvent` to keep things at one place

File: pkg/ethereum/node_notifiee.go [hunk 8]
```diff
@@ -109,13 +107,6 @@ func (n *Node) handleOutboundConnection(pid peer.ID) {
...
-
```
Comment (namn-grg) [inline pkg/ethereum/node_notifiee.go:118]: Doing this inside `sendMetadataEvent` to keep things at one place

**Review by namn-grg — COMMENTED** [review]
@mempirate ready for review!

File: config/config.go (hunk 3 above)
Comment (mempirate) [reply config/config.go]: Let's rename it to the table names, i.e. `discovery_events.log`

File: config/config.go (hunk 4 above)
Comment (mempirate) [reply config/config.go]: rename to `metadata_events.log`

**Review by mempirate — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 4 above)
Comment (namn-grg) [reply config/config.go]: done!

File: config/config.go (hunk 3 above)
Comment (namn-grg) [reply config/config.go]: done!

**Pull request merged**